### PR TITLE
Stats page of Settings menu

### PR DESCRIPTION
**Pull request by pusztaienike**

### CODE REVIEW

File: apps/sensenet/src/components/settings/stats-components-widget.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+export const ComponentsWidget: React.FunctionComponent<ComponentsWidgetProps> = (props) => {
+  const classes = useStyles()
+  const widgetClasses = useWidgetStyles()
+  const localization = useLocalization().settings
+
+  return (
+    <div className={widgetClasses.root}>
+      <Paper elevation={0} className={widgetClasses.container}>
+        <div className={classes.rowContainer}>
+          <span>{localization.components}</span>
```
Comment (taki9) [inline apps/sensenet/src/components/settings/stats-components-widget.tsx:40]: remove span

File: apps/sensenet/src/components/settings/stats-components-widget.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+                <TableCell align="left">{localization.lastOfficialVersion}</TableCell>
+                <TableCell align="left">{localization.description}</TableCell>
+              </TableRow>
+            </TableHead>
+            <TableBody>
+              {props.data.Components.map((row) => (
+                <TableRow key={row.ComponentId}>
+                  <TableCell align="left">{row.ComponentId}</TableCell>
+                  <TableCell align="left">{row.Version}</TableCell>
+                  <TableCell align="left">{localization.na}</TableCell>
```
Comment (taki9) [inline apps/sensenet/src/components/settings/stats-components-widget.tsx:57]: I would not render that column

File: apps/sensenet/src/components/settings/stats-installed-packages-widget.tsx [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+  const globalClasses = useGlobalStyles()
+  const widgetClasses = useWidgetStyles()
+  const localization = useLocalization().settings
+  const dateUtils = useDateUtils()
+
+  return (
+    <div className={widgetClasses.root}>
+      <Paper elevation={0} className={widgetClasses.container}>
+        <div className={classes.rowContainer}>
+          <span>{localization.installedPackages}</span>
```
Comment (taki9) [inline apps/sensenet/src/components/settings/stats-installed-packages-widget.tsx:48]: remove span

File: apps/sensenet/src/components/settings/stats-storage-widget.tsx [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+
+  return (
+    <div className={widgetClasses.root}>
+      <Paper elevation={0} className={widgetClasses.container}>
+        <div className={classes.rowContainer}>
+          <span>{localization.storage}</span>
+          <span>
+            {localization.used(
+              `${numberFormatter.format(round(allUsage / 1024 / 1024))} MB`,
+              `${numberFormatter.format(round(data.subscription.plan.limitations.storage / 1024))} GB`,
```
Comment (taki9) [inline apps/sensenet/src/components/settings/stats-storage-widget.tsx:61]: is it good to compare values with different units?

File: apps/sensenet/src/components/settings/stats-storage-widget.tsx [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+          backgroundColor={theme.palette.type === 'light' ? theme.palette.action.disabled : theme.palette.common.white}
+          visualParts={[
+            {
+              percentage: `${
+                allUsage > data.usage.storage.available
+                  ? (data.usage.storage.files / allUsage) * 100
+                  : (data.usage.storage.files / data.usage.storage.available) * 100
+              }%`,
+              color: '#57c1f7',
+              title: 'Files',
```
Comment (taki9) [inline apps/sensenet/src/components/settings/stats-storage-widget.tsx:75]: title should be translatable

File: apps/sensenet/src/components/settings/stats.tsx [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+        idOrPath: '/Root',
+        name: 'GetVersionInfo',
+        method: 'GET',
+      })
+
+      setData(response)
+    })()
+  }, [repository])
+
+  if (!data) return null
```
Comment (taki9) [inline apps/sensenet/src/components/settings/stats.tsx:29]: GetVersionInfo and GetDashboardData requests are nonparallel this way, causing longer data loading.

All the data loading should happen here with a loading indicator and render the whole page at once.

**Review by taki9 — COMMENTED** [review]
I think block titles (Storage, Components, Installed packages) should be more emphasized some way.

File: apps/sensenet/src/components/settings/stats-storage-widget.tsx (hunk 4 above)
Comment (pusztaienike) [reply apps/sensenet/src/components/settings/stats-storage-widget.tsx:61]: Check the design: https://miro.com/app/board/o9J_klNk0R4=/

**Comment by pusztaienike** [comment]
> I think block titles (Storage, Components, Installed packages) should be more emphasized some way.

Can you please give me details? 
https://miro.com/app/board/o9J_klNk0R4=/
On the design it has the same font size and type like other texts.

**Review by herflis — APPROVED** [review]

File: packages/sn-controls-react/src/fieldcontrols/file-size.tsx [hunk 7]
```diff
@@ -14,6 +14,20 @@ import { defaultLocalization } from './localization'
...
+}
+
+export const returnValueWithUnit = (fieldValueNumber: number, index = 0): string => {
+  const inHigherUnit = round(fieldValueNumber / 1024)
+  if (inHigherUnit >= 1 && units.length > index + 1) {
+    return returnValueWithUnit(inHigherUnit, index + 1)
+  } else {
+    return `${fieldValueNumber} ${units[index]}`
+  }
+}
```
Comment (taki9) [inline packages/sn-controls-react/src/fieldcontrols/file-size.tsx:29]: move it to helpers.ts

rename: returnValueWithUnit -> formatSize

**Review by taki9 — APPROVED** [review]

**Pull request merged**